woocommerce/woocommerce-android
# First iteration on the implementation of CardReaderPaymentViewModel

Parent issue #3726 

Note: Half of the changes are unit tests which imo help with the review so I'm ignoring the "changes > 500" peril check. Having said that, let me know if you think I should still split it into multiple PRs.

This PR implements a first iteration of logic in CardReaderPaymentViewModel. The ViewModel is responsible for starting the "payment" flow and updating ViewState based on the events received from CardReaderManager. The view layer doesn't listen to ViewState events yet. I've also added a temporary ShowSnackBar events which display the received events in a snackbar - just for testing purposes.

There is also an update in the PaymentManager - the received amount is converted from dollars to cents, since the StripeTerminalSDK accepts only currency smallest unit. (I should have made this change in a separate PR, just let me know and I can split it)

Note: The current implementation doesn't work with other currency but USD.

To Test
- Store with working WCPay is required to test these changes
1. Open app settings
2. Tap on "Manager card reader"
3. (optional) Check "use simulated card reader"
4. Tap "Connect to reader"
4. Go back to homepage
5. Open detail of an order with USD currency
6. Tap on "Collect payment" button
7. Notice a snackbar with current state is shown (tap your card if you are using hw reader)

Update release notes:
